common/Geometry2d/Rect.hpp

```diff
@@ -76,6 +76,9 @@ class Rect : public Shape {
     bool nearSegment(const Segment& seg, float threshold) const;
 
     bool intersects(const Rect& other) const;
+    bool intersects_(const Segment& other) const;
+
+    int CohenSutherlandCode(const Point& other) const;
```

Comment:
This function can be private since it shouldn't be used anywhere else

Reply:
changed